articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+
+```bash
+git clone https://github.com/clonehad/vows.git
+
+cd vows
+
+npm install
+```
+
+## 2. Understand the A to Z Most Useful Node.js Modules and Frameworks
+
+The power of Node.js is backed by hundreds of packages that make it easier to write your program. These packages are open sourced and they are the ecosystem of Node.js applications. Every Node.js developer will interact with these packages to get a more efficient development workflow.
+
+Here's a list of some of the most useful libraries that will help ease the development of Node.js applications.
+
+-   [Express](http://expressjs.com/): is a web application framework for Node.js that handles robust APIs and Web servers.
+
+-   [Gulp](https://www.npmjs.com/package/gulp): helps to automate development workflow. It helps to automate simple tasks in the development of web applications.
+
+-   [Moment](https://www.npmjs.com/package/moment): is an insubstantial JavaScript date library for parsing, validating, manipulating, and formatting dates.
+
+-   [Async](https://caolan.github.io/async/v3/): is a utility module that provides powerful straightforward functions for working with asynchronous JavaScript.
+
+-   [Lodash](https://lodash.com/): makes JavaScript easier by taking out the irritation of working with arrays, numbers, objects, strings, etc.
+
+-   [JSHint](https://www.npmjs.com/package/jshint): detects errors and potential problems in your Node.js (JavaScript) code. You can easily adjust it in th environment you expect your code to execute.
+
+-   [Cheerio](https://cheerio.js.org/): Commonly used for web page scraping. It implements the subset of jQuery and removes `DOM`inconsistency and browser cleverness to extract API endpoint URLs from web pages.
+
+-   [Browserify](https://github.com/browserify/browserify): analyses `require ()` calls in your Node.js application to create a bundle that you can serve the browser in a single `<script>` tag.
+
+-   [Nodemailer](https://nodemailer.com/about/): Allows (easy as cake) email sending.
+
+You can have a look at more available libraries on [GitHub](https://github.com/aravindnc/A-to-Z-List-of-Useful-Node.js-Modules).
+
+## 3. The Node.js HTTP Server Module
+
+Node.js. has a built-in module called HTTP (Hypertext Transfer Protocol). It helps in making requests as a client and running the server to accept requests and return responses. 
+
+In this case, when you visit a webpage from a browser, you make a request to another computer (webserver) via the internet, which then provides you the webpage response. The webserver receives HTTP requests from the client, which
+listens to a server port and provides an HTTP response for example a HTML Page. 
+
+To use HTTP and client server in Node.js you must use `require ('http');`. Let's look at a simple Hello World HTTP Server module that will listen on port number 3000.
+```js
+var http = require('http')
```

Comment:
Tried to run this code block and received an error: `C:\Users\Owner\Downloads\api-nodejs-demo\index.js:5
                    text / plain " });
                    ^^^^

SyntaxError: Unexpected identifier`